### PR TITLE
[CST-2657] Fix timeout in mailer query

### PR DESCRIPTION
**Pull request by tonyheadford**
### Context

- Ticket: [Jira](https://dfedigital.atlassian.net/browse/CST-2687)

Refactor query to require cohort and also try to make dry run counts more accurate in the other bulk mailers

### Changes proposed in this pull request
Make cohort a required param for the query.
Add the join clauses before calling `.count` in the bulk mailer methods.

### CODE REVIEW

File: app/services/bulk_mailers/school_reminder_comms.rb [hunk 1]
```diff
@@ -73,12 +73,13 @@ def contact_sits_that_need_to_appoint_an_ab_for_unregistered_ects
...
 
+      query = query.joins(:induction_coordinator_profiles)
+
       return query.count if dry_run
 
       email_count = 0
 
       query
-        .joins(:induction_coordinator_profiles)
         .includes(:induction_coordinator_profiles)
```
Comment (ltello) [inline app/services/bulk_mailers/school_reminder_comms.rb:83]: The query has  both, joins and includes:
 `.joins(:induction_coordinator_profiles).includes(:induction_coordinator_profiles)` no?

File: app/services/bulk_mailers/school_reminder_comms.rb [hunk 2]
```diff
@@ -117,14 +119,15 @@ def contact_sits_that_have_not_added_participants
...
+
       return query.count if dry_run
 
       email_count = 0
 
       # this could send a lot of email at the cohort start and may break Notify limits
       # numbers should be checked before running this (dry_run = true) and maybe changes made/different approach to batch these
       query
-        .joins(:induction_coordinator_profiles)
         .includes(:induction_coordinator_profiles)
```
Comment (ltello) [inline app/services/bulk_mailers/school_reminder_comms.rb:131]: same here

Comment (tonyheadford) [reply app/services/bulk_mailers/school_reminder_comms.rb]: As above.

File: app/services/bulk_mailers/school_reminder_comms.rb (hunk 1 above)
Comment (tonyheadford) [reply app/services/bulk_mailers/school_reminder_comms.rb]: Yes, I'm using `.includes` to eager load not to left outer join but will switch to `.eager_load` to make the intention clear. Old habits.

**Review by ltello — APPROVED** [review]

**Pull request merged**